exporters/trace/jaeger/agent_test.go

```diff
@@ -99,3 +103,49 @@ func TestNewAgentClientUDPWithParamsReconnectingDisabled(t *testing.T) {
 
 	assert.NoError(t, agentClient.Close())
 }
+
+type errorHandler struct{ t *testing.T }
+
+func (eh errorHandler) Handle(err error) { assert.NoError(eh.t, err) }
+
+func TestJaegerAgentUDPLimitBatching(t *testing.T) {
+	otel.SetErrorHandler(errorHandler{t})
+
+	// 1500 spans, size 79559, does not fit within one UDP packet with the default size of 65000.
+	n := 1500
+	s := make([]*tracesdk.SpanSnapshot, n)
+	for i := 0; i < n; i++ {
+		s[i] = &tracesdk.SpanSnapshot{}
+	}
+
+	exp, err := NewRawExporter(
+		WithAgentEndpoint(WithAgentHost("localhost"), WithAgentPort("6831")),
+	)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	assert.NoError(t, exp.ExportSpans(ctx, s))
+	assert.NoError(t, exp.Shutdown(ctx))
+}
+
+func TestSpanExceedsMaxPacketLimit(t *testing.T) {
+	otel.SetErrorHandler(errorHandler{t})
+
+	// 106 is the serialized size of a span with default values.
+	maxSize := 106
+	span := &tracesdk.SpanSnapshot{
+		Name: "a-longer-name-that-makes-it-exceeds-limit",
+	}
+	largeSpans := []*tracesdk.SpanSnapshot{span, span, {}}
```

Comment:
Is this triggering the failure because there are two spans or because the span is too big?

Can this be changed back to one large span make it clear what this test is testing?

Reply:
> Is this triggering the failure because there are two spans or because the span is too big?
> 
> Can this be changed back to one large span make it clear what this test is testing?

This line doesn't cause the test failure, i change this line to increase test coverage; Also i add a single large span test case that will explain what this test is testing.
The compatibility-test for windows actually causes the test failure, the span serialization bytes on windows is 4 bytes larger than linux/mac, it will make the normal span(empty ones) produces an error when exported, i change the maxSize to elimate the error.

Reply:
I'm aware this line does not produce the test failure, but it is [a change that was included](https://github.com/open-telemetry/opentelemetry-go/pull/1853/commits/ce591ba699a32afa39ebd07da55de918e4ec4f94) in the problem you are describing that is not related to the failure you mention:

```diff
-	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
+	oneLargeSpan := []*tracesdk.SpanSnapshot{span}
+	largeSpans := []*tracesdk.SpanSnapshot{span, span, {}}
```

Having `largeSpans` now include 2 `span` makes it less clear what we are testing. Are we testing if two large spans should overload the UDP packet size when included with an empty span, or one?

```suggestion
	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
```

Reply:
> I'm aware this line does not produce the test failure, but it is [a change that was included in the fix commited](https://github.com/open-telemetry/opentelemetry-go/pull/1853/commits/ce591ba699a32afa39ebd07da55de918e4ec4f94) to addrss the problem you are describing that is does not seem related to the failure:
> 
> ```diff
> -	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
> +	oneLargeSpan := []*tracesdk.SpanSnapshot{span}
> +	largeSpans := []*tracesdk.SpanSnapshot{span, span, {}}
> ```
> 
> Having `largeSpans` now include 2 `span` makes it less clear what we are testing. Are we testing if two large spans should overload the UDP packet size when included with an empty span, or one?

The reason why i put two large `span` in the `largeSpans` slice is very simple: to produce mutiple errors, and it will cover the joined error string logic. `oneLargeSpan` and `largeSpans` are two different cases,representing one single error and mutiple errors when processing a batch.

Reply:
The issue I'm talking about is that this test is testing multiple things. Please split the different errors cases you are testing into easily recognized test cases. This could mean splitting into multiple tests, refactoring into generalized testing conditions, or, at the very least, document what is being tested.

Reply:
OK, I'll put them in different test cases